### ISSUE TITLE
Using oauthenticator with F5 Big-IP APM

### ISSUE THREAD

**Issue by guillaumeeb** [jupyterhub/oauthenticator#259]
Hi everyone,

Has anybody ever tried to plug oauthenticator to F5 Big-IP APM. I'm trying to do this (with a colleague of mine, I know nothing about oauth), but did not managed to yet.

This idea is to use GenericOAuthenticator directly, with a config similar to this:
```yaml
c.OAuthenticator.client_id = 'id_hash'
c.OAuthenticator.client_secret = 'secret_hash'
c.GenericOAuthenticator.token_url = 'https://myF5hostname/f5-oauth2/v1/token'
c.GenericOAuthenticator.userdata_url = 'https://myF5hostname/f5-oauth2/v1/introspect'
#c.GenericOAuthenticator.userdata_url = 'https://myF5hostname/f5-oauth2/v1/token'
c.GenericOAuthenticator.userdata_method = 'POST'
c.GenericOAuthenticator.userdata_params = {"grant_types": "client_credentials"}
#c.GenericOAuthenticator.userdata_params = {"id_token": "access_token"}
c.GenericOAuthenticator.username_key = "preferred_username"
c.GenericOAuthenticator.oauth_callback_url = "https://jupyterhub-qualif.sis.cnes.fr/hub/oauth_callback"
```

So the difficulties I have is to find the correct userdata_url (there is no userdata endpoint with F5 version 13.x), and the correct parameters to give. No success so far... I get the token, but don't manage to query the username. So if someone has any insight, that would be really nice.

**Comment by dmpe** [comment]
@guillaumeeb Do you have any update on this ? Did you manage to get it work with F5? Maybe their support could help...

**Comment by guillaumeeb** [comment]
Hi @dmpe. I was not able to make it work with oauth, but we made it kind of work with SAML: https://github.com/bluedatainc/jupyterhub-samlauthenticator.

We stopped for a time as we lack some configuration on the F5 side, but we'll be working again on it soon, and I've good hope that it'll work fine. That's one of the many things I'll need to share on the open, probably through Pangeo means, see https://discourse.pangeo.io/t/how-about-creating-a-pangeo-hpc-github-repo/150.
